services/api/service.go

```diff
@@ -835,6 +924,71 @@ func (api *RelayAPI) handleGetPayload(w http.ResponseWriter, req *http.Request)
 		if err != nil {
 			log.WithError(err).Error("failed to increment builder-stats after getPayload")
 		}
+
+		builderStatus, err := api.redis.GetBlockBuilderStatus(bidTrace.BuilderPubkey.String())
+		if err != nil {
+			log.WithError(err).Error("failed to get builder status")
+		}
+
+		// Check if the block was valid if the builder is OptimisticActive or OptimisticDemoted.
+		if builderStatus == common.OptimisticActive || builderStatus == common.OptimisticDemoted {
+			// Set to locked while we process the winning block.
+			api.setStatusByCollateralID(bidTrace.BuilderPubkey.String(), common.OptimisticLocked)
```

Comment:
There may be a more fundamental problem: we can't rely on `handleGetPayload` to determine the winning block. The reason is that the validator could collude with the builder to get the payload out of band, without ever calling `handleGetPayload`. This could even happen without collusion when multiple relays forwarded the same header and the proposer gets the payload from another relay first.

Reply:
as discussed offline, i don't see this as being a significant issue. if `handleGetPayload` is never called, then we never publish the block, thus we aren't responsible for refunding the proposer even if the block turns out to be invalid. i don't think it is possible for a proposer to trick us into thinking they deserve a refund unless we publish a signed, invalid block. if that is the case and they are colluding with the builder to make sure they get a bad block, we refund them from the builders collateral anyway, so it is just their own money changing hands. let me know if i am missing something here!